fs-storage/src/file_storage.rs

```diff
@@ -1,58 +1,115 @@
-use std::fmt::Debug;
 use std::fs::{self, File};
 use std::io::{BufRead, BufReader, BufWriter, Write};
-use std::str::FromStr;
 use std::time::SystemTime;
 use std::{
     collections::BTreeMap,
     path::{Path, PathBuf},
 };
 
+use crate::base_storage::BaseStorage;
 use data_error::{ArklibError, Result};
 
 const STORAGE_VERSION: i32 = 2;
 const STORAGE_VERSION_PREFIX: &str = "version ";
 
-pub struct FileStorage {
+pub struct FileStorage<K, V> {
     label: String,
     path: PathBuf,
     timestamp: SystemTime,
+    data: BTreeMap<K, V>,
 }
 
-impl FileStorage {
+impl<K, V> FileStorage<K, V>
+where
+    K: Ord + Clone + serde::Serialize + serde::de::DeserializeOwned,
+    V: Clone + serde::Serialize + serde::de::DeserializeOwned,
+{
     /// Create a new file storage with a diagnostic label and file path
     pub fn new(label: String, path: &Path) -> Self {
-        Self {
+        let mut file_storage = Self {
             label,
             path: PathBuf::from(path),
             timestamp: SystemTime::now(),
+            data: BTreeMap::new(),
+        };
+
+        // Load the data from the file
+        file_storage.data = match file_storage.read_fs() {
+            Ok(data) => data,
+            Err(err) => {
+                log::error!("Error reading storage file: {}", err);
```

Comment:
Indeed, the first time a FileStorage is created the file doesn't exist. Perhaps that can be checked here. If the file doesn't exist when creating the FileStorage it's being created for the first time. So it can start with an empty BTreeMap.